### PR TITLE
add 'get_ref()' and 'get_mut()' to 'Accept'

### PR DESCRIPTION
**Pull request by SergioBenitez**
Fairly straightforward. I need a way to access the stream, to pull data from it, while in the `Accept` phase.

### CODE REVIEW

**Comment by quininer** [comment]
It sounds weird to access io in the `Accept` phase, can you describe the scenario?

**Comment by SergioBenitez** [comment]
See the server example at https://github.com/rustls/hyper-rustls/blob/main/examples/server.rs. Without this change, there's no way to access the connection in the Accept phase in either AsyncRead or AsyncWrite. See also my specific use case in https://github.com/SergioBenitez/Rocket/commit/9b3437b52b1e04b0ad3964528f46bba73d4aabe4.

**Comment by quininer** [comment]
In my opinion this use case is more about satisfying abstractions than providing actual features.
It doesn't make much sense to change `nodelay` or call `poll_flush` during Accept phase, but I'm a little unsure about `poll_shutdown`.

I don't mind adding these methods, it's reasonable to add, but easy to abuse.

File: tokio-rustls/src/lib.rs [hunk 1]
```diff
@@ -315,6 +315,22 @@ impl<IO> Accept<IO> {
     pub fn into_fallible(self) -> FallibleAccept<IO> {
         FallibleAccept(self.0)
     }
+
+    pub fn get_ref(&self) -> Option<&IO> {
```
Comment (quininer) [inline tokio-rustls/src/lib.rs:319]: for symmetry, can we also add it to `Connect`?

Comment (SergioBenitez) [reply tokio-rustls/src/lib.rs:335]: Done.

**Comment by SergioBenitez** [comment]
> In my opinion this use case is more about satisfying abstractions than providing actual features.
> It doesn't make much sense to change `nodelay` or call `poll_flush` during Accept phase, but I'm a little unsure about `poll_shutdown`.

I generally agree, though seeing as usage with `hyper` is likely a primary use-case of this crate, I think this is a concession that must unfortunately be made.

**Review by quininer — APPROVED** [review]
LGTM

**Comment by SergioBenitez** [comment]
Any chance we could get this in a point release soon?

**Comment by djc** [comment]
See #105.

**Pull request merged**